### ISSUE TITLE
SkipTest by default

### ISSUE THREAD

**Issue by a1339398734** [catboost/catboost#541]
Problem: {
Just following the tutorial in [https://github.com/catboost/catboost/blob/master/catboost/tutorials/classification/classification_tutorial.ipynb](url), but find a interesting phenomenon in the code bellow:

> tunned_model = CatBoostClassifier(
    random_seed=63,
    iterations=1000,
    learning_rate=0.03,
    l2_leaf_reg=3,
    bagging_temperature=1,
    random_strength=1,
    one_hot_max_size=2,
    leaf_estimation_method='Newton'
)
tunned_model.fit(
    X_train, y_train,
    cat_features=cat_features,
    verbose=100,
    eval_set=(X_validation, y_validation),
    plot=True
)

Here is the output of the above code:

> MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))
0:	learn: 0.6569860	test: 0.6570642	best: 0.6570642 (0)	total: 70.1ms	remaining: 1m 10s
100:	learn: 0.1697219	test: 0.1530068	best: 0.1530068 (100)	total: 3.7s	remaining: 33s
200:	learn: 0.1585173	test: 0.1407452	best: 0.1407452 (200)	total: 8.38s	remaining: 33.3s
300:	learn: 0.1553371	test: 0.1382992	best: 0.1382986 (299)	total: 12.8s	remaining: 29.7s
400:	learn: 0.1532426	test: 0.1375255	best: 0.1375234 (398)	total: 18s	remaining: 26.9s
500:	learn: 0.1512429	test: 0.1368889	best: 0.1368856 (499)	total: 23.7s	remaining: 23.6s
600:	learn: 0.1498213	test: 0.1367448	best: 0.1366783 (566)	total: 28.9s	remaining: 19.2s
700:	learn: 0.1482981	test: 0.1364651	best: 0.1364651 (700)	total: 34.4s	remaining: 14.7s
800:	learn: 0.1471757	test: 0.1364096	best: 0.1364017 (797)	total: 39.7s	remaining: 9.87s
900:	learn: 0.1459933	test: 0.1364278	best: 0.1362697 (850)	total: 45.2s	remaining: 4.97s
999:	learn: 0.1450301	test: 0.1364453	best: 0.1362697 (850)	total: 50.5s	remaining: 0us
bestTest = 0.1362696814
bestIteration = 850
Shrink model to first 851 iterations.

The best validation score is 0.1362696814 and the best iteration is 850.
Then I import the log_loss from sklearn to measure the prediction:

> from sklearn.metrics import log_loss
print(log_loss(y_validation, tunned_model.predict_proba(X_validation)[:, 1]))

the result is 0.1362696814001939, same as the previous.

Then I make a little change to the model: 1. Change the iterations from 1000 to 850(which is the best iteration); 2. comment out the eval_set in fit function, here is the model after change:

> tunned_model = CatBoostClassifier(
    random_seed=63,
    iterations=850,
    learning_rate=0.03,
    l2_leaf_reg=3,
    bagging_temperature=1,
    random_strength=1,
    one_hot_max_size=2,
    leaf_estimation_method='Newton'
)
tunned_model.fit(
    X_train, y_train,
    cat_features=cat_features,
    verbose=100,
    plot=True)

And I rerun the mode, the result is show bellow:

> MetricVisualizer(layout=Layout(align_self='stretch', height='500px'))
0:	learn: 0.6569860	total: 42.2ms	remaining: 35.8s
100:	learn: 0.1700584	total: 3.75s	remaining: 27.8s
200:	learn: 0.1603529	total: 8.01s	remaining: 25.9s
300:	learn: 0.1570913	total: 12.1s	remaining: 22s
400:	learn: 0.1546539	total: 16.4s	remaining: 18.3s
500:	learn: 0.1527133	total: 21.4s	remaining: 14.9s
600:	learn: 0.1510568	total: 26.7s	remaining: 11s
700:	learn: 0.1497814	total: 31.4s	remaining: 6.68s
800:	learn: 0.1484412	total: 36.8s	remaining: 2.25s
849:	learn: 0.1480436	total: 39.2s	remaining: 0us

same as before, I use the log_loss from sklean to measure the prediction of model, but the result change from 0.1362696814001939 to 0.13892971272961754, which make me confused, as I know that the eval_set is used for overfitting detection, whether use eval_set will not affect the prediction result if I use the same number of iterations, it would be very appreciated that someone can help me figure out this 
problem.

}

catboost version: {0.11.1}
Operating System: {Ubuntu 18.04.1 LTS}
CPU: {Intel® Core™ i7-7600U CPU @ 2.80GHz × 4 }

**Comment by annaveronika** [comment]
Eval set is also used for counting frequencies of categorical features. For example, if you have categorical feature animal, then if your train dataset has 5 cats and validation dataset has 2 cats the resulting value of the feature will be 7. This doesn't lead to overfitting, because you don't use labels of the validation dataset, in fact you can set labels of validation dataset to all 0, and quality will not change.

If you want to disable it, you should use counter_calc_method='SkipTest'

We will disable this by default, because we are tired of answering this question :)
In one of the next releases this will be an additional feature.

**Comment by annaveronika** [comment]
And I'm happy to see that people are using the tutorials!

**Comment by a1339398734** [comment]
@annaveronika Thank your for your answer. BTW the tutorials is awesome;)

**Comment by kizill** [comment]
We finally switched default counter calculation method to `SkipTest` in 58de256b12e5a2b75bbb838e56a8167251ea6d8d
